pkg/networkmanager/firewall.jsx

```diff
@@ -199,13 +202,19 @@ class AddServicesModal extends React.Component {
     }
 
     save() {
+        let p;
         if (this.state.custom) {
-            firewall.createService(this.state.custom_id, this.state.custom_name, this.createPorts())
+            p = firewall.createService(this.state.custom_id, this.state.custom_name, this.createPorts())
                     .then(firewall.enableService(this.state.zones, this.state.custom_id));
         } else {
-            firewall.addServices(this.state.zones, [...this.state.selected]);
+            p = firewall.addServices(this.state.zones, [...this.state.selected]);
         }
-        this.props.close();
+        p.then(() => this.props.close()).catch(error => {
```

Comment:
Again, please move the .catch to the next line, so that this doesn't visually apply to the ".then" part.